.github/workflows/pypi_publish.yaml

```
@@ -16,10 +16,24 @@ jobs:
         with:
           python-version: "3.9"
 
-      - name: Verify Package Version vs Tag Version
+      - name: Ensure package version and tag are equal
+        working-directory: ${{ inputs.working_directory }}
+        shell: bash
         run: |
-          PKG_VER="$(grep -oP 'version = "\K[^"]+' pyproject.toml)"
-          TAG_VER="${GITHUB_REF##*/}"
+          if [ -f "setup.py" ]
+          then
+            PKG_VER="$(python setup.py --version)"
+          elif [ -f "pyproject.toml" ]
+          then
+            version=$(python3.11 -c "import tomllib; print(tomllib.load(open('pyproject.toml', 'rb'))['project']['version'])")
```

```suggestion
            version=$(python3 -c "import tomli; print(tomli.load(open('pyproject.toml', 'rb'))['project']['version'])")
```
We set up python with version 3.9 above, no need to hardcode a version number here.
Which also means, we don't have tomllib available, as that's only part of the stdlib from 3.11 onward.
Alternatively, you  could install tomli, as that's what we use in e.g. the `update_pyproject` script